backend/main.go

```diff
@@ -352,7 +398,29 @@ func main() {
 			return
 		}
 	}
+
 	server := Server{store: store, storeTimeout: defaultStoreTimeout}
+
+	if authMode == "none" {
+		server.auth = noAuth{}
+	} else if authMode == "firebase" {
+		log.Printf("Using firebase authentication per command-line flag")
+		ctx := context.Background()
+		app, err := firebase.NewApp(ctx, nil)
+		if err != nil {
+			log.Fatalf("Could not instantiate Firebase app: %v\n", err)
+			return
+		}
+		firebaseClient, err := app.Auth(ctx)
+		if err != nil {
+			log.Fatalf("Could not get Firebase Auth client: %v\n", err)
+			return
+		}
+
+		firebaseAuth := firebaseAuth{firebaseClient: firebaseClient}
+		server.auth = firebaseAuth
+	}
```

Comment:
We should have an `else` here that calls `log.Fatalf` to tell the user they requested an unsupported `authMode`.

Reply:
Good point, done.